support/build/Makefile.rules

```diff
@@ -540,6 +540,21 @@ $(eval $(call vprefix_lib,$(1),OBJS-y) += $(4))
 $(eval $(call vprefix_lib,$(1),CLEAN-y) += $(call build_clean,$(4)) $(call out2dep,$(4)))
 endef
 
+define buildrule_rs =
+$(4): $(2) | preprocess
+	$(call build_cmd,RUSTC,$(1),$(4),\
+		$(RUSTC) $$(RUSTCFLAGS) $$(RUSTCFLAGS-y) \
```

Comment:
Can we also include the general compiler flags that we have, like the optimization flags, flags for ISR code and the file-variant specific flags?
Is `rustc` GCC flag compatible?

Reply:
Rustc is not GCC compatible, I've adapted the debug and optimization flags accordingly.

Reply:
Are there includes that need to be considered for `rustc`?
Can `rustc` generate something for dependency management?